### PR TITLE
Feature: create file explorer component

### PR DESCRIPTION
**Pull request by justinehell**

### CODE REVIEW

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/composables/FileExplorer.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+/**
+ * Convert fileInformation input to the Object required to display Quasar Tree Component
+ * @param {Array} fileInformation - Object to store all file information.
+ * Example: [{path: '/home/folderA/file1.tf'}, {path: '/home/folderA/file2.tf'}]
+ * @returns {Array} - The array of nodes that designates the tree structure of Quasar Tree Component
+ * @see https://quasar.dev/vue-components/tree
+ */
+export function getTree(fileInformation) {
+  const tree = [];
+  fileInformation.forEach((pathObject) => {
```
Comment (Zorin95670) [inline src/composables/FileExplorer.js:76]: `fileInformation.forEach((pathObject)` -> `fileInformations.forEach((fileInformation)`

File: src/composables/FileExplorer.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import { randomHexString } from 'src/composables/Random';
+
+/**
+ * Create and add a new folder.
+ * @param {Array} folder - Folder that will receive an empty folder Object
+ * @param {String} pathName - Name of the new folder.
+ */
+function createFolder(folder, pathName) {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/FileExplorer.js:8]: Done

File: src/composables/FileExplorer.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  }
+  return folders;
+}
+
+/**
+ * Sort Tree elements by alphabetic order, then sort to display folders before files
+ * @param {Array} arr - Tree folders
+ * @returns {Array} - Sorted Tree
+ */
+function sortTree(arr) {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/FileExplorer.js:53]: Done

File: src/composables/FileExplorer.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    isFolder: true,
+  });
+}
+
+/**
+ * Create and add a new file.
+ * @param {Array} folder - Folder that will receive an empty file Object
+ * @param {String} fileName - Name of the new folder.
+ */
+function createFile(folder, fileName) {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/FileExplorer.js:23]: Done

File: src/composables/FileExplorer.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      return sortTree(node.children);
+    }
+    return arr;
+  });
+  return arr;
+}
+
+/**
+ * Convert fileInformation input to the Object required to display Quasar Tree Component
+ * @param {Array} fileInformation - Object to store all file information.
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/FileExplorer.js:69]: ✅

File: src/composables/FileExplorer.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+ * Example: [{path: '/home/folderA/file1.tf'}, {path: '/home/folderA/file2.tf'}]
+ * @returns {Array} - The array of nodes that designates the tree structure of Quasar Tree Component
+ * @see https://quasar.dev/vue-components/tree
+ */
+export function getTree(fileInformation) {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/FileExplorer.js:74]: ✅

File: src/composables/FileExplorer.js (hunk 1 above)
Comment (justinehell) [reply src/composables/FileExplorer.js:76]: ✅

File: src/components/FileExplorer.vue [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    :nodes="nodes"
+    node-key="label"
+    :no-nodes-label="$t('actions.fileExplorer.empty')"
+    dense
+    icon="fa-solid fa-chevron-right"
+  >
+    <template #default-header="{expanded, node}">
+      <div class="tree-node">
+        <q-icon color="primary" :name="`${node.icon}${expanded ? '-open' : ''}`" class="q-mx-sm" />
+        <span @click="selected = node.id" :class="{ 'text-bold' : selected === node.id}">
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/FileExplorer.vue:12]: ✅

File: src/components/FileExplorer.vue [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  <q-tree
+    :nodes="nodes"
+    node-key="label"
+    :no-nodes-label="$t('actions.fileExplorer.empty')"
+    dense
+    icon="fa-solid fa-chevron-right"
+  >
+    <template #default-header="{expanded, node}">
+      <div class="tree-node">
+        <q-icon color="primary" :name="`${node.icon}${expanded ? '-open' : ''}`" class="q-mx-sm" />
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/FileExplorer.vue:11]: ✅

File: src/components/FileExplorer.vue [hunk 9]
```diff
@@ -0,0 +1,38 @@
+<template>
+  <q-tree
+    :nodes="nodes"
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/FileExplorer.vue:3]: ✅

File: tests/unit/components/ModelizerTextView.spec.js [hunk 10]
```diff
@@ -5,6 +5,10 @@ import ModelizerTextView from 'src/components/ModelizerTextView.vue';
 installQuasarPlugin();
 
 describe('Test component: ModelizerTextView', () => {
+  window.crypto = {
```
Comment (Zorin95670) [inline tests/unit/components/ModelizerTextView.spec.js:8]: remove it

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/components/ModelizerTextView.vue [hunk 11]
```diff
@@ -1,24 +1,38 @@
...
 
-<style lang="scss" scoped>
-.modelizer-text-view {
-  display: flex;
-  flex: 1;
-  flex-direction: column;
-  height: 100vh;
-}
-</style>
+// TODO : Hard-coded data to display FileEplorer component
```
Comment (k-y0u) [inline src/components/ModelizerTextView.vue:23]: the TODO should be "remove hard-coded data when ... "

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/composables/FileExplorer.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
+/**
+ * Create and add a new folder.
+ * @param {Array} folder - Folder that will receive an empty folder Object
+ * @param {String} name - Name of the new folder.
+ */
+function createFolder(id, folder, name) {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:6]: done

File: src/composables/FileExplorer.js [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+    isFolder: true,
+  });
+}
+
+/**
+ * Create and add a new file.
+ * @param {Array} folder - Folder that will receive an empty file Object
+ * @param {String} name - Name of the new folder.
+ */
+function createFile(id, folder, name) {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:21]: done

File: tests/unit/components/ModelizerTextView.spec.js (hunk 10 above)
Comment (Zorin95670) [reply tests/unit/components/ModelizerTextView.spec.js:8]: done

File: src/composables/FileExplorer.js [hunk 14]
```diff
@@ -0,0 +1,92 @@
+/**
+ * Create and add a new folder.
+ * @param {Array} folder - Folder that will receive an empty folder Object
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:3]: done

File: src/composables/FileExplorer.js [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+    icon: 'fa-solid fa-folder',
+    label: name,
+    children: [],
+    isFolder: true,
+  });
+}
+
+/**
+ * Create and add a new file.
+ * @param {Array} folder - Folder that will receive an empty file Object
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:18]: done

File: src/composables/FileExplorer.js [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+    icon: 'fa-regular fa-file',
+    label: name,
+    isFolder: false,
+  });
+}
+
+/**
+ * Return folder children from a given path
+ * @param {Array} folders - Tree folders
+ * @param {String[]} path - Array of path
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:33]: done

File: src/composables/FileExplorer.js [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+  folder.push({
+    id,
+    icon: 'fa-regular fa-file',
+    label: name,
+    isFolder: false,
+  });
+}
+
+/**
+ * Return folder children from a given path
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:31]: done

File: src/composables/FileExplorer.js [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+ * @param {Array} folders - Tree folders
+ * @param {String[]} path - Array of path
+ */
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:34]: done

File: src/composables/FileExplorer.js [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+    .sort((x, y) => ((x.isFolder === y.isFolder) ? 0 : x.isFolder ? -1 : 1));
+
+  elements.filter((element) => element.children && element.children.length > 0)
+    .forEach((element) => sortTreeElements(element.children));
+  return elements;
+}
+
+/**
+ * Convert fileInformation input to the Object required to display Quasar Tree Component
+ * @param {FileInformation[]} fileInformations - Array to convert for Quasar Tree Component
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/FileExplorer.js:63]: done

File: src/components/ModelizerTextView.vue (hunk 11 above)
Comment (Zorin95670) [reply src/components/ModelizerTextView.vue:23]: done

**Review by k-y0u — DISMISSED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Review by Zorin95670 — DISMISSED** [review]

**Review by k-y0u — DISMISSED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**